### PR TITLE
task 11

### PR DESCRIPTION
**Pull request by annc4st**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -392,6 +392,41 @@ describe('GET /api/users', () =>{
...
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({body}) => {         
+            expect(body.articles).toHaveLength(12)
+            body.articles.forEach((article) => {
+                expect(article.topic).toEqual('mitch')
+            })
+        })
+    });
+    test('GET 200 if topic does not exist sends array of all articles',  () => {
```
Comment (robert112233) [inline __tests__/app.test.js:420]: I would send back a 404 to the client, as seeing all the articles wouldn't be very clear 👍

File: models/models.js [hunk 2]
```diff
@@ -44,19 +44,40 @@ exports.fetchArticleById = (id) => {
...
-ON c.article_id = a.article_id
-GROUP BY a.article_id
-ORDER BY a.created_at DESC; 
-`;
+//5 and 11 sortby
+exports.fetchArticles = (topic) => {
+  const validTopics = {
+    mitch: 'mitch',
+    cats: 'cats'
+  };
```
Comment (robert112233) [inline models/models.js:52]: What if you expand the API in the future to allow clients to add their own articles? Hardcoding it here is not the best way to keep track of the valid articles

Comment (robert112233) [reply models/models.js:52]: You could use a topics db query to get the valid topics

**Review by robert112233 — CHANGES_REQUESTED** [review]
Amend the faulty test and merge without review, great work 👍

File: models/models.js (hunk 2 above)
Comment (annc4st) [reply models/models.js:52]: added query to look for available topics

File: __tests__/app.test.js (hunk 1 above)
Comment (robert112233) [reply __tests__/app.test.js:420]: Nice work, don't forget to change the description on line 420

File: __tests__/app.test.js [hunk 3]
```diff
@@ -392,6 +392,54 @@ describe('GET /api/users', () =>{
...
+describe('GET /api/articles/:article_id', () => {
+    test('GET 200 resonds with article object where coment_count is correct', () => {
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({body}) => {    
+      
+            expect(body.article.article_id).toBe(1)
+            expect(body.article.title).toBe("Living in the shadow of a great man" )
+            expect(body.article.comment_count).toBe(11)
```
Comment (robert112233) [inline __tests__/app.test.js:439]: Nicely done

**Review by robert112233 — APPROVED** [review]
Update the test description then Merge! Nice work 🥇

**Pull request merged**